File: src/main/java/com/server/ordering/domain/dto/response/OrderDetailDto.java
package com.server.ordering.domain.dto.response;

import com.server.ordering.domain.Order;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;
import java.util.stream.Collectors;

import static lombok.AccessLevel.PROTECTED;

@Getter @Setter
@AllArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class OrderDetailDto extends OrderPreviewWithRestSimpleDto {

    List<OrderFoodDto> orderFoods;

    public OrderDetailDto(Order order) {
        super(order);
        this.orderFoods = order.getOrderFoods().stream().map(OrderFoodDto::new).collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/dto/response/OrderDetailDto.kt
package com.byeoru.ordering_server.domain.dto.response

import com.byeoru.ordering_server.domain.Order

data class OrderDetailDto(val order: Order) {

    val orderFoods: List<OrderFoodDto>

    init {
        this.orderFoods = order.orderFoods.map { OrderFoodDto(it) }
    }
}